### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by kkavod**

### CODE REVIEW

File: frontend/src/app/components/channel/channel.singleUser.controller.ts [hunk 1]
```diff
@@ -41,20 +41,26 @@ export class SingleUserChannelController extends BaseChannelController {
...
+        attachMediaStream(this.slotElement.program, this.programUser.stream);
+
+        if (oldProgramUser) {
+          this.videoRoom.unsubscribeFromStream(oldProgramUser.login);
+          oldProgramUser.stream = null;
+          this.$log.debug('Unsubscribed from', oldProgramUser.login);
+        }
+      }, () => {
+        // Rolling back current program user in case of error
+        this.programUser = oldProgramUser;
```
Comment (edoshor) [inline frontend/src/app/components/channel/channel.singleUser.controller.ts]: How would we know this happens ?
May be log.error(...)

File: frontend/src/app/components/channel/channel.singleUser.controller.ts [hunk 2]
```diff
@@ -65,22 +71,28 @@ export class SingleUserChannelController extends BaseChannelController {
...
+        this.previewUser = user;
+
+        if (oldPreviewUser && oldPreviewUser !== this.programUser) {
+          this.videoRoom.unsubscribeFromStream(oldPreviewUser.login);
+          oldPreviewUser.stream = null;
+          this.$log.debug('Unsubscribed from', oldPreviewUser.login);
+        }
+
+      }, () => {
+        this.toastr.error(`Unable to subscribe for user ${user.login}`);
```
Comment (edoshor) [inline frontend/src/app/components/channel/channel.singleUser.controller.ts]: How would we know this happens ?
May be log.error(...)

Comment (kkavod) [inline frontend/src/app/components/channel/channel.singleUser.controller.ts]: it does call it inside this.videoRoom.subscribeForStream before propagating to here
add one more report here?

File: frontend/src/app/components/channel/channel.singleUser.controller.ts (hunk 1 above)
Comment (kkavod) [inline frontend/src/app/components/channel/channel.singleUser.controller.ts]: it does call it inside forwardProgramToSDI before propagating to here
add one more report here?

File: frontend/src/app/components/channel/channel.singleUser.controller.ts (hunk 2 above)
Comment (edoshor) [inline frontend/src/app/components/channel/channel.singleUser.controller.ts]: No. it's ok for now

**Pull request merged**